rcm_nexus/command.py

```diff
@@ -57,79 +83,88 @@ def push(repo, environment, product, version, ga=False, debug=False):
     More information: https://mojo.redhat.com/docs/DOC-1199638
     """
 
-    nexus_config = config.load(environment, debug=debug)
-
-    session = Session(nexus_config, debug=debug)
-    print("Pushing: %s content to: %s" % (repo.decode("utf-8"), environment))
-
-    try:
-        # produce a set of clean repository zips for PUT upload.
-        zips_dir = tempfile.mkdtemp()
-        print("Creating ZIP archives in: %s" % zips_dir)
-        if os.path.isdir(repo):
-            print("Processing repository directory: %s" % repo)
-
-            # Walk the directory tree, and create a zip.
-            zip_paths = archive.create_partitioned_zips_from_dir(repo, zips_dir)
-        else:
-            print("Processing repository zip archive: %s" % repo)
+    nexus_config = config.load(environment)
+    npm_archive_type = npm.detect_npm_archive(repo)
+    if npm_archive_type != npm.NpmArchiveType.NOT_NPM:
```

Comment:
The logic / language here is hard to follow. It would be better to make the `else` logic the `if` and undo some of the negation in this condition, then put this logic into the `else` section.

Additionally, it seems like we're not considering splitting the NPM archive up according to some maximum upload size. I think this could be risky, once we start shipping more (and more complex) products, particularly since NPM packages can stuff all kinds of crazy things into the tarball.

Reply:
This unfortunately had to be done due to Python 2.x support, that does not process enums the same way as 3.x in boolean expressions. The original if was 
if npm_archive_type:
   blah blah

This was much more clear. Unfortunately 0 enum value is not treated as False in Python 2. Thus I had to explicitly use the not equal. I could comment it though.